File: eu_scraper.rb
#!/usr/bin/env ruby

require 'bundler/setup'
require 'rubygems'

require 'net/http'
require 'nokogiri'
require 'open-uri'
require 'fileutils'

year = 2015
document_type = 'QECR'   # Parliamentary Written Questions
language = 'EN'   # Language => English

max_page_num = 1
current_page = 1

# Create docs/ directory if it doesn't exist
FileUtils::mkdir_p 'docs'

# Trap Ctrl-C/SIGINT signal
interrupted = false
trap("INT") { puts "Shutting Down!"; interrupted = true }

loop do
  uri = URI.parse("http://www.europarl.europa.eu/RegistreWeb/search/typedoc.htm?codeTypeDocu=#{document_type}&year=#{year}&lg=#{language}&currentPage=#{current_page}")
  response = Net::HTTP.get(uri)
  page = Nokogiri::HTML(response)

  # Update New Max Page Number
  max_page_num = page.css(".ep_paginate li a").last.attr("title").to_i

  # Get Links to Questions on the current page
  links = page.css(".results .notice .documents a").map { |record| record.attr("href") }

  # Download Files to docs/ directory
  links.each do |link|
    file_name = link.split('/').last
    File.open("docs/#{file_name}", "wb") do |f|
      f << open(link).read
    end

    # If Process has been interrupted or is completed, exit the program
    exit if interrupted
  end

  # If no more documents remain to be downloaded
  break if current_page == max_page_num
end

puts "Process Successfully Completed"

File: eu-scraper.py
#!/usr/bin/env python
"""
    Requirements:
    - requests (installation: pip install requests)
    - lxml (installation: pip install lxml)
"""

import requests
import lxml.html
import os

def download_file(file_name, url):
    #file_name = url.split('/')[-1]
    # NOTE the stream=True parameter
    r = requests.get(url, stream=True)
    with open(file_name, 'wb') as f:
        for chunk in r.iter_content(chunk_size=1024):
            if chunk:
                f.write(chunk)
    return file_name


for i in range(1, 1381):
    url = "http://www.europarl.europa.eu/RegistreWeb/search/typedoc.htm?codeTypeDocu=QECR&year=2015&currentPage={0}".format(i)
    html = lxml.html.parse(url)
    titles = [i.strip() for i in html.xpath("//div[contains(@class, 'notice')]/p[@class='title']/a/text()")]
    docs = [i.strip() for i in html.xpath("//div[contains(@class, 'notice')]/ul/li/a/@href")]
    q_refs = [i.strip() for i in html.xpath("//div[contains(@class, 'notice')]/div[@class='date_reference']/span[2]/text()")]
    for title, doc, q_ref in zip(titles, docs, q_refs):
        file_name = os.path.join(os.getcwd(),'data','-'.join(title.split('/'))+' '+q_ref+'.'+doc.split('.')[-1])
        downloaded_file = download_file(file_name, doc)
        print downloaded_file